app/components/promise-spinner.js

```diff
@@ -11,23 +14,26 @@ export default Ember.Component.extend({
 
   retrieveData: function () {
     var url = this.get('url');
-    var self = this;
-    this.set('isPending', true);
-    Ember.$.ajax({
-      url: url
-    }).then(function (data) {
-      self.setProperties({
-        isPending: false,
-        promiseValue: data ? data : "No price found"
+    var promiseSpinnerUrls = this.get('promiseSpinnerUrls');
+    var dataFound = promiseSpinnerUrls.hasData(url);
+    var promiseController;
+
+    if (dataFound) {
+      promiseController = PromiseController.create({
+        promise: new Ember.RSVP.Promise(function (resolve, reject) {
```

Comment:
No need to change it, but I came across this syntax the other day - `Ember.RSVP.resolve(dataFound);`